### ISSUE TITLE
[RFC] Handle language slug as subdomain

### ISSUE THREAD

**Issue by zomars** [AnimalDesign/bolt-translate#95]
Is it possible to handle the language routing from `http://domain.com/es/` to `http://es.domain.com/`?

Or some like this in the [Symfony](http://symfony.com/doc/current/routing/hostname_pattern.html) docs.

Cheers.

**Comment by madc** [comment]
@SahAssar just did something with different domains, so it should be possible.

**Comment by zomars** [comment]
Is that true @SahAssar ?

**Comment by SahAssar** [comment]
@zomars Yeah, It's not 100% tested yet though, and I'll have to check the modifications I had to make to the urlgenerator (I think, it's a bit hazy), but this is the gist of it (for a finnish/swedish site where I used TLD's instead of subdomains):

```
homepage:
    path: /
    host: "www.yourdomain.{_locale}"
    defaults:
        _controller: controller.frontend:homepage
        _locale: sv
    requirements:
        _locale: "sv|fi"

search:
    path: /search
    host: "www.yourdomain.{_locale}"
    defaults:
        _controller: controller.frontend:search
        _locale: sv
    requirements:
        _locale: "sv|fi"

preview:
    path: /preview/{contenttypeslug}
    host: "www.yourdomain.{_locale}"
    defaults:
        _controller: controller.frontend:preview
        _locale: sv
    requirements:
        contenttypeslug: controller.requirement:anyContentType
        _locale: "sv|fi"

contentlink:
    path: /{contenttypeslug}/{slug}
    host: "www.yourdomain.{_locale}"
    defaults:
        _controller: controller.frontend:record
        _locale: sv
    requirements:
        contenttypeslug: controller.requirement:anyContentType
        _locale: "sv|fi"

taxonomylink:
    path: /{taxonomytype}/{slug}
    host: "www.yourdomain.{_locale}"
    defaults:
        _controller: controller.frontend:taxonomy
        _locale: sv
    requirements:
        taxonomytype: controller.requirement:anyTaxonomyType
        _locale: "sv|fi"
```

And then in my menu:

```
main:
    - label: Start
      path: homepage
    - label: Produkter
      link: '#'
      submenu:
        - label: Hemsida
          route: pagebinding
          param:
              slug: Ny-hemsida
        - label: Webshop
          route: pagebinding
          param:
              slug: Webshop
        - label: Sökmotoroptimering
          route: pagebinding
          param:
              slug: Sökmotoroptimering
        - label: Logotyp
          route: pagebinding
          param:
              slug: Logodesign
    - label: Referenser
      route: pagebinding
      param:
          slug: Referenser
    - label: Kundservice
      route: pagebinding
      param:
          slug: Kundservice
    - label: Kontakt
      route: pagebinding
      param:
          slug: Kontakta-Oss
```

Lemme know if it works for you and we can add it to the docs.

**Comment by zomars** [comment]
This looks awesome @SahAssar, I will try it ASAP. Unrelated question, are you handling multiple TLD in a single bolt instalation? Or is it a multisite? I'm curious now...

**Comment by SahAssar** [comment]
@zomars Well, this particular site is a single bolt install with just the TLDs nginx config pointing to the same root. Part of the changes I made in https://github.com/bolt/bolt/pull/6209 was to allow it to detect the canonical domain for each language without having to set it explicitly.

On our normal sites it's all a single multisite install, so that's basically every site having it's on config files, database file and theme folder, but all the php files are shared by setting the SCRIPT_FILENAME fastcgi param to a single php file that bootstraps bolt for each of the individual sites.

**Comment by SahAssar** [comment]
@zomars Just checking in, did that routing worrk out for you?

**Comment by SahAssar** [comment]
Just tried it out, it required one small tweak in the localized menu builder (fixed in c4a6e7e64f0ecbad5121869a1cde8e3b4df1557e), and I added the instructions to the readme.

**Comment by zomars** [comment]
Yes It was working with some minor caveats, will test this new version and let you know.
